File: main/contrib/Windows API Code Pack 1.1/source/Samples/Shell/StarBackupUsingLibraries/CS/Wizard/WizardLauncher.cs
//Copyright (c) Microsoft Corporation.  All rights reserved.

using System;
using System.Windows;
using System.Windows.Controls;
using System.Windows.Navigation;

namespace Microsoft.WindowsAPICodePack.Samples.StarBackupSample
{
    public class WizardLauncher : PageFunction<WizardResult>
    {
        public event WizardReturnEventHandler WizardReturn;

        protected override void Start()
        {
            base.Start();

            // So we remember the WizardCompleted event registration
            this.KeepAlive = true;
            
            // Launch the wizard
            StarBackupMain StarBackupMain = new StarBackupMain();
            StarBackupMain.Return += new ReturnEventHandler<WizardResult>(wizardPage_Return);
            this.NavigationService.Navigate(StarBackupMain);
        }

        public void wizardPage_Return(object sender, ReturnEventArgs<WizardResult> e)
        {
            // Notify client that wizard has completed
            // NOTE: We need this custom event because the Return event cannot be
            // registered by window code - if WizardDialogBox registers an event handler with
            // the WizardLauncher's Return event, the event is not raised.
            if (this.WizardReturn != null)
            {
                this.WizardReturn(this, new WizardReturnEventArgs(e.Result, null));
            }
            OnReturn(null);
        }
    }
}

File: main/contrib/Windows API Code Pack 1.1/source/Samples/Shell/StarBackupUsingLibraries/VB/Wizard/WizardLauncher.vb
'Copyright (c) Microsoft Corporation.  All rights reserved.


Imports Microsoft.VisualBasic
Imports System
Imports System.Windows
Imports System.Windows.Controls
Imports System.Windows.Navigation

Namespace Microsoft.WindowsAPICodePack.Samples.StarBackupSample
	Public Class WizardLauncher
		Inherits PageFunction(Of WizardResult)
		Public Event WizardReturn As WizardReturnEventHandler

		Protected Overrides Sub Start()
			MyBase.Start()

			' So we remember the WizardCompleted event registration
			Me.KeepAlive = True

			' Launch the wizard
			Dim StarBackupMain As New StarBackupMain()
			AddHandler StarBackupMain.Return, AddressOf wizardPage_Return
			Me.NavigationService.Navigate(StarBackupMain)
		End Sub

		Public Sub wizardPage_Return(ByVal sender As Object, ByVal e As ReturnEventArgs(Of WizardResult))
			' Notify client that wizard has completed
			' NOTE: We need this custom event because the Return event cannot be
			' registered by window code - if WizardDialogBox registers an event handler with
			' the WizardLauncher's Return event, the event is not raised.
			If Me.WizardReturnEvent IsNot Nothing Then
				RaiseEvent WizardReturn(Me, New WizardReturnEventArgs(e.Result, Nothing))
			End If
			OnReturn(Nothing)
		End Sub
	End Class
End Namespace
